subprojects/core/build.gradle.kts

```diff
@@ -182,11 +183,18 @@ dependencies {
         because("Some tests utilise the 'java-gradle-plugin' and with that TestKit")
     }
     crossVersionTestDistributionRuntimeOnly(project(":distributions-core"))
+
+    annotationProcessor(project(":internal-instrumentation-processor"))
```

Comment:
If I understand correctly, we now run an annotation processor when compiling `core`. What kind of performance impact has this? Is the annotation processor incremental?

Reply:
Your understanding is correct. One the tool is adapted to the needs of the Execution team (properties transformation), are going want to also run this annotation processor (maybe with a different set of "extensions") in some other subprojects. 
I have not measured the performance.
Semantically, it is aggregating processor, but in terms of the APIs, some of the processed annotations only have the source retention for now, which we can change. It is therefore not an incremental processor yet in terms of APIs.

Reply:
Do we have any performance tests for the `gradle/gradle` build?

Reply:
```
Created classpath snapshot for incremental compilation in 0.006 secs.
...
Class dependency analysis for incremental compilation took 0.105 secs.
```
Any better way to get the incremental compilation state report than these lines in `--info`?

Reply:
On my machine, the annotation processing time reported in the build scan is 0.183s. The full run time of `:core:compileJava`, though, degrades significantly, from around 1s to nearly 6s, which might well mean a full recompilation. 🙁

Reply:
> ```Full recompilation is required because org.gradle.internal.instrumentation.processor.ConfigurationCacheInstrumentationProcessor is not incremental```

Reply:
Yes, it should be logged on the info level, something like `Full recompilation is required because...`. Yeah, I noticed the processor is not incremental, but I am not sure if it's a simple change to get that in 8.1.

Since there is a hurry to get this in 8.1,  maybe we could move instrumentation classes out of core to some separate project and add that instrumentation project as a dependency to launcher (something similar as I did for code-quality project).

Reply:
So this change is necessary for stable configuration cache?

Note that I don't care too much if the incrementality makes it in 8.1 or not. What I don't want is that Gradle's engineers suffer for more than a week of full recompilation of core.

Reply:
> So this change is necessary for stable configuration cache?

Yes, it is.

I made the annotation processor incremental in https://github.com/gradle/gradle/pull/24165/commits/fb9caebebcbc1397bb194de25756308f333f4dc4, and now the `:core:compileJava` runs incrementally and in approximately the same time as without the processor.

Reply:
Awesome, that is great!